editor/groups_editor.cpp

```diff
@@ -295,19 +305,26 @@ void GroupDialog::_rename_group_item(const String &p_old_name, const String &p_n
 	}
 }
 
-void GroupDialog::_load_groups(Node *p_current) {
-	List<Node::GroupInfo> gi;
-	p_current->get_groups(&gi);
+void GroupDialog::_load_groups(Node *p_current, HashMap<StringName, bool> &p_groups_map) {
```

Comment:
This should be `r_groups_map`  as per my above comment.

Reply:
Hmm, but does it really make sense here?
Since I'm not reassigning the `HashMap`, but just inserting values.

Reply:
I agree with @Maran23 on that, the reference is not being written to, the referenced object is just being modified (and that's fine/expected, it's not being passed as `const &`).

In the same manner e.g. `Object::get_property_list` has its parameter named `p_list` because it's not creating/returning a new list, it just populates the list passed in.

Reply:
There are some cases in the codebase where we use the r_ prefix for such return parameters(see `AStarGrid2D::_get_nbors(Point *p_point, List<Point *> &r_nbors)` or  `DynamicBVH::_fetch_leaves(Node *p_root, LocalVector<Node *> &r_leaves, int p_depth)` for example), although it's not consistent.

In #33027 we agreed upon:
> This convention is that method parameters should start with a `p_` prefix (like "parameter"), unless they are modified by the function itself, in which case they start with a `r_` prefix (like "return" - BTW if anyone knows the proper technical jargon to refer to this kind of parameters, that would be useful for the docs). This convention makes it easy to differentiate function parameters from local variables and members, and to spot parameters which might change state and be further used in their calling method.

Reply:
The question here what count as 'modification'. Only a reassignment, or more?
For a reassignment I agree, but in case we are modifiying values as well (like inserting)?

We are modifying `p_` variables all over the place. Consider the following example:
```
fit_child_in_rect(Control *p_child) {
    p_child->set_rect(r);
}
```
This is a modification, but we still name it with `p_` (and if this is wrong, we need to rename a lot of references in the code...)

Reply:
Checking other recently merged PRs like https://github.com/godotengine/godot/pull/70994 from reduz, no `r_` is used. In this case for a `HashSet` which is filled by a method. So this is fine as long as I do not reassign the `HashMap`.